buildRelease.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+usage(){
+cat<<EOF
+
+ Tag & build a DBS2Go release
+ Usage:
+   buildRelease.sh [options]
+
+ options:
+ -r <remote>       Remote repository to update, default to origin.
+ -b <gitbranch>    Git branch to be tagged, default to master.
+ -t <tag>          Tag name to be created.
+ -h                Show this helper.
+
+
+EOF
+}
+
+# Set defaults:
+REMOTE='origin'
+GITBRANCH='master'
+TAG=
+
+set -e
+
+while getopts ":r:b:t:h" opt; do
+    case ${opt} in
+        r)
+            REMOTE=$OPTARG ;;
+        b)
+            GITBRANCH=$OPTARG ;;
+        t)
+            TAG=$OPTARG ;;
+        h)
+            usage; exit 0 ;;
+        \? )
+            echo -e "\nERROR: Invalid Option: -$OPTARG\n"
+            usage
+            exit 1 ;;
+        : )
+            echo -e "\nERROR: Invalid Option: -$OPTARG requires an argument\n"
+            usage
+            exit 1 ;;
+    esac
+done
+# shift to the last parsed option, so we can consume the non-parametric
+# arguments passed to the call with a regular shift
+# shift $(expr $OPTIND - 1 )
+
+[[ -z $TAG ]] && { usage ; echo "ERROR: no TAG provided"; exit 3 ;}
+
+# Check whether we are in the correct directory
+[[ $(git rev-parse --show-toplevel) == $PWD ]] || {
+  echo "ERROR: Not in root directory $(git rev-parse --show-toplevel)!"
+  exit 5
+}
+
+# Check if tag exists
+git show-ref --tags --quiet -- $TAG && {
+  echo "ERROR: Tag $TAG exists!"
+  exit 6
+}
+
+echo "Checking out branch: $GITBRANCH"
+git checkout $GITBRANCH
+
+echo "Pulling from upstream"
+git pull $REMOTE $GITBRANCH
+git fetch $REMOTE
+
+# Fetch the last tag's commit line (hash id plus tag name)
+tagRegExp="^v[0-9]+[0-9]+\.[0-9]+[0-9]+\.[0-9]+[0-9]+\.*(rc|patch)*[0-9]*$"
+LASTCOMMITLINE=$(git log -n1 --oneline --no-decorate -E --grep=$tagRegExp)
+LASTCOMMIT=$(echo ${LASTCOMMITLINE} | awk '{print $1}')
+LASTVERSION=$(echo ${LASTCOMMITLINE} | awk '{print $2}')
+
+echo "Generating CHANGES file"
+TMP_CHANGES=$(mktemp -t wmcore.${LASTVERSION}.XXXXX)
```

Comment:
Todor, since you cut and paste this script please walk through any occurence of wmcore prefixes and change them accordingly to dbs2go or something not related to wmcore.

Reply:
Hi @vkuznet  I was not done yet :)  with my latest commit I rewrote quite a bunch of stuff  in this script. The script is currently completely agnostic to the repository it is working on. Once I finish with the dbs2go Makefile changes I'll call for your review again... basically in few minutes I hope...